src/Modal.js

```diff
@@ -315,4 +349,37 @@ const Modal = React.createClass({
   }
 });
 
-export default Modal;
+const ModalOverlay = React.createClass({
+  propTypes: {
+    ...Portal.propTypes
+  },
+
+  defaultProps: {
+    show: null
+  },
+
+  render() {
+    let { show, ...props } = this.props;
+
+    let modal = (
+      <Modal {...props}>{this.props.children}</Modal>
+    );
+    // I can't think of another way to not break back compat while defaulting container
+    if ( show != null ){
+      return (
+        <Portal show={show} container={props.container} >
+          { modal }
+        </Portal>
+      );
+    } else {
+      return modal;
+    }
+  }
+});
+
+ModalOverlay.Body = Body;
```

Comment:
It's weird thing. It seems a little weird to set things up this way - if you were using just ES6 modules, you'd be doing named exports, and they'd still have to be imported separately.

@mtscout6 You mentioned you were considering something similar for the dropdown overhaul. Which approach are you taking for e.g. `DropdownButtonTitle`?